include/openthread/platform/radio.h

```diff
@@ -621,6 +622,32 @@ bool otPlatRadioGetPromiscuous(otInstance *aInstance);
  */
 void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable);
 
+/**
+ * Indicates whether radio should stay in Receive or Sleep state during idle periods.
```

Comment:
Honestly I spent some time to understand this change. So I think maybe it's easier to understand if putting like this:
```suggestion
 * Sets the rx-on-when-idle state to the radio platform.
 * 
 * There are a few situations that the radio can enter sleep state if the 
 * device is in rx-off-when-idle state but it's hard and costly for the SubMac
 * to identify these situations and instruct the radio to enter sleep:
 * 
 * - Finalization of a regular frame reception task, provided that:
 *   - The frame is received without errors and passes the filtering and it's not an spurious ACK.
 *   - ACK is not requested or transmission of ACK is not possible due to internal conditions.
 * - Finalization of a regular frame transmission task or transmission of an ACK frame, when ACK is not requested in
 *   the transmitted frame.
 * - Finalization of the reception operation of a requested ACK due to:
 *   - ACK timeout expiration.
 *   - Reception of an invalid ACK or not an ACK frame.
 *   - Reception of the proper ACK, unless the transmitted frame was a Data Request Command and the frame pending bit
 *     on the received ACK is set to true.
 * - Finalization of a stand alone CCA task.
 * - Finalization of a CCA operation with busy result during CSMA/CA procedure.
 * - Finalization of an Energy Detection task.
 *
* If the radio platform has the capability to enter sleep state in these
* situations, it will enter sleep when the state is set to rx-off-when-idle.
```

In addition, shall we add a otError return value which will be `OT_ERROR_NOT_SUPPORTED` if the radio platform doesn't support it?

Reply:
Not sure if adding the error helps here, since the API won't be called anyway if the radio doesn't support `OT_RADIO_CAPS_RX_ON_WHEN_IDLE`.

Reply:
SGTM. We can keep the void return.